plotutil/xyer.go

```diff
@@ -0,0 +1,77 @@
+package plotutil
+
+import "github.com/gonum/plot/plotter"
+
+type xyer struct {
+	x, y []float64
+	len  int
+}
+
+// Len returns the length of the XYer object.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) Len() int {
+	return xy.len
+}
+
+// XY returns the x,y values at index idx.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) XY(idx int) (x, y float64) {
+	return xy.x[idx], xy.y[idx]
+}
+
+// ZipXY is a convenience function to build a plotter.XYer from slices.
+// Length will be set to the shortest slice length.
+// Panics on empty or nil slices.
+func ZipXY(x, y []float64) plotter.XYer {
```

Comment:
Another option is to do 

````
type VecXY struct {
   X, Y []float64
}

func (v VecXY) Len() int {
    if len(v.X) != len(v.Y) { 
        panic("length mismatch")
   }
   return len(x)
}

func (v VecXY) XY(i int) (x, y float64){
    return x[i], y[i]
}
````

This shifts where the panic happens, but I don't think that's a huge problem. It removes having an unexported type as part of a package, which gets tricky for things like json. This is easy enough to construct, plotutil.VecXY{x,y}, and it's not clear to me that we really need to panic in the len == 0 case.  Name can be modified obviously.